Community mod manager support field (TS-2279)

Roffenlund opened the pull request:
- Add a new boolean field to the Community model to indicate whether a community supports mod manager functionality.

- Use this field to control the visibility of the "Install with Mod Manager" button on the Package Detail page.

- Implement all necessary properties and tests to ensure proper functionality.

MythicManiac commented on django/thunderstore/community/models/community.py:
```diff
@@ -104,6 +104,9 @@ class Community(TimestampMixin, models.Model):
     )
 
+    # Will hide/show "Install with Mod Manager" button on package pages
+    mod_manager_support = models.BooleanField(default=True)
```
We've used a naming convention where boolean fields are generally prefixed with `is_` or `has_` etc to make it more clear, could do `has_mod_manager_support` here for example.

MythicManiac reviewed (commented):
We could also add the new field as a list filter option on the django admin for communities

Roffenlund commented:
@MythicManiac 

Updated with the following changes:
- Changed `Community` model field name from `mod_manager_support` -> `has_mod_manager_support`.
- Removed old migration and generated new one.
- Updated `has_mod_manager_support` property in `PackageListing`.
- Add `has_mod_manager_support` to community admin filters and added the column to the table.
- Updated tests where necessary.

MythicManiac reviewed (changes requested):
The migration ID 0032 is already used (due to another PR being merged), so the migration should probably be updated in this PR before merging
